pkg/webhook/elasticquota/quota_topology.go

```diff
@@ -0,0 +1,664 @@
+
+	quotaName := qt.getQuotaNameFromPodNoLock(pod)
+	qt.podCache[quotaName][pod.Name] = struct{}{}
+	klog.V(5).Infof("onPodAdd, link pod :%v to quota:%v", pod.Name, quotaName)
+}
+
+func (qt *quotaTopology) OnPodUpdate(oldObj, newObj interface{}) {
+	oldPod := oldObj.(*v1.Pod)
+	newPod := newObj.(*v1.Pod)
+
+	qt.lock.Lock()
+	defer qt.lock.Unlock()
+
+	oldQuotaName := qt.getQuotaNameFromPodNoLock(oldPod)
+	newQuotaName := qt.getQuotaNameFromPodNoLock(newPod)
+	delete(qt.podCache[oldQuotaName], oldPod.Name)
+	qt.podCache[newQuotaName][newPod.Name] = struct{}{}
+}
+
+func (qt *quotaTopology) OnPodDelete(obj interface{}) {
+	pod := obj.(*v1.Pod)
+
+	qt.lock.Lock()
+	defer qt.lock.Unlock()
+
+	quotaName := qt.getQuotaNameFromPodNoLock(pod)
+	delete(qt.podCache[quotaName], pod.Name)
+}
+
+func (qt *quotaTopology) AddQuota(quota *v1alpha1.ElasticQuota) (err error) {
+	if quota == nil {
+		return fmt.Errorf("AddQuota param is nil")
+	}
+
+	qt.lock.Lock()
+	defer qt.lock.Unlock()
+
+	if _, exist := qt.quotaInfoMap[quota.Name]; exist {
+		return fmt.Errorf("quota already exist:%v", quota.Name)
+	}
+
+	// fill quota with basic Information
+	err = qt.fillDefaultQuotaInfo(quota)
+	if err != nil {
+		klog.Errorf("AddQuota fillDefaultQuotaInfo failed:%v", err)
+		return err
+	}
+
+	err = qt.basicItemCheck(quota)
+	if err != nil {
+		klog.Errorf("AddQuota basicItemCheck failed :%v", err)
+		return err
+	}
+
+	quotaInfo := core.NewQuotaInfoFromQuota(quota)
+
+	err = qt.validateQuota(quotaInfo)
+	if err != nil {
+		klog.Errorf("AddQuota validateQuota failed:%v", err)
+		return err
+	}
+
+	qt.quotaInfoMap[quotaInfo.Name] = quotaInfo
+	qt.podCache[quotaInfo.Name] = make(map[string]struct{})
+	if _, ok := qt.quotaHierarchyInfo[quotaInfo.Name]; !ok {
+		qt.quotaHierarchyInfo[quotaInfo.Name] = make(map[string]struct{})
+	}
+	if quotaInfo.ParentName != extension.RootQuotaName {
+		qt.quotaHierarchyInfo[quotaInfo.ParentName][quotaInfo.Name] = struct{}{}
+	}
+	return nil
+}
+
+func (qt *quotaTopology) UpdateQuota(oldQuota, originQuota *v1alpha1.ElasticQuota) (err error) {
+	if originQuota == nil {
+		return fmt.Errorf("AddQuota param is nil")
+	}
+
+	if oldQuota != nil && reflect.DeepEqual(quotaFieldsCopy(oldQuota), quotaFieldsCopy(originQuota)) {
+		return nil
+	}
+
+	quotaName := originQuota.Name
+	qt.lock.Lock()
+	defer qt.lock.Unlock()
+	localQuotaInfo, exist := qt.quotaInfoMap[quotaName]
+	if !exist {
+		err = fmt.Errorf("quota not exist:%v", quotaName)
+		klog.Errorf("UpdateQuota failed:%v", err)
+		return err
+	}
+
+	err = qt.basicItemCheck(originQuota)
+	if err != nil {
+		klog.Errorf("UpdateQuota basicItemCheck failed %v", err)
+		return err
+	}
+
+	newQuotaInfo := core.NewQuotaInfoFromQuota(originQuota)
+	// quota 信息检查
+	err = qt.validateQuota(newQuotaInfo)
+	if err != nil {
+		klog.Errorf("UpdateQuota validateQuota failed:%v", err)
+		return err
+	}
+
+	// quota 层次关系发生改变
+	err = qt.checkQuotaHierarchyHasChange(localQuotaInfo, newQuotaInfo)
```

Comment:
why CreateQuota not need checkQuotaHierarchyHasChange? what if newQuota's parent not exist? what if add a newQuota leads to SumMin too large? 

I think you don't think clear clear with create\update Quota,  these two process shouldn't has any different at all